xtext/editorserver/Servlet.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * generated by Xtext 2.29.0
+ */
+package DSLQNAME;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.nio.charset.Charset;
+import java.util.stream.Collectors;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.eclipse.epsilon.egl.EglTemplateFactoryModuleAdapter;
+import org.eclipse.epsilon.emc.emf.InMemoryEmfModel;
+import org.eclipse.xtext.resource.IResourceServiceProvider;
+import org.eclipse.xtext.resource.XtextResourceSet;
+import org.eclipse.xtext.util.DisposableRegistry;
+import org.eclipse.xtext.web.server.InvalidRequestException;
+import org.eclipse.xtext.web.servlet.XtextServlet;
+
+import com.google.gson.Gson;
+import com.google.inject.Injector;
+
+import net.sourceforge.plantuml.FileFormat;
+import net.sourceforge.plantuml.FileFormatOption;
+import net.sourceforge.plantuml.SourceStringReader;
+
+/**
+ * Deploy this class into a servlet container to enable DSL-specific services.
+ */
+@WebServlet(name = "XtextServices", urlPatterns = "/xtext-service/*")
+public class DSLNAMEServlet extends XtextServlet {
+
+    private static final long serialVersionUID = 1L;
+
+    DisposableRegistry disposableRegistry;
+
+    public void init() throws ServletException {
+        super.init();
+        Injector injector = new DSLNAMEWebSetup().createInjectorAndDoEMFRegistration();
+        this.disposableRegistry = injector.getInstance(DisposableRegistry.class);
+    }
+
+    private class ToXMIResponse {
+        public String output;
+    }
+
+    private class ErrorResponse {
+        public String error;
+    }
+
+    private class ToDiagramResponse {
+        public String diagram;
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
```

Comment:
The logic that's been added could be refactored out into a separate class that's called here in doPost() to make it easier to understand.

Reply:
Ok agree